src/main/java/io/supertokens/webserver/api/oauth/OAuthClientsAPI.java

```diff
@@ -0,0 +1,168 @@
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.webserver.api.oauth;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.supertokens.Main;
+import io.supertokens.multitenancy.exception.BadPermissionException;
+import io.supertokens.oauth.OAuth;
+import io.supertokens.oauth.exceptions.OAuthClientException;
+import io.supertokens.oauth.exceptions.OAuthClientRegisterException;
+import io.supertokens.oauth.exceptions.OAuthException;
+import io.supertokens.pluginInterface.RECIPE_ID;
+import io.supertokens.pluginInterface.Storage;
+import io.supertokens.pluginInterface.exceptions.InvalidConfigException;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.multitenancy.AppIdentifier;
+import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.io.Serial;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class OAuthClientsAPI extends WebserverAPI {
+
+    @Serial
+    private static final long serialVersionUID = -4482427281337641246L;
+
+    private static final List<String> ALLOWED_INPUT_FIELDS = Arrays.asList(new String[]{"clientName","scope", "redirectUris", "allowedCorsOrigins", "authorizationCodeGrantAccessTokenLifespan", "authorizationCodeGrantIdTokenLifespan", "authorizationCodeGrantRefreshTokenLifespan",
+        "clientCredentialsGrantAccessTokenLifespan","implicitGrantAccessTokenLifespan","implicitGrantIdTokenLifespan","refreshTokenGrantAccessTokenLifespan","refreshTokenGrantIdTokenLifespan","refreshTokenGrantRefreshTokenLifespan","tokenEndpointAuthMethod","audience",
+        "grantTypes","responseTypes","clientUri","logoUri","policyUri","tosUri","metadata"});
+    private static final List<String> REQUIRED_INPUT_FIELDS = Arrays.asList(new String[]{"clientName", "scope"});
+
+    @Override
+    public String getPath() {
+        return "/recipe/oauth/clients";
+    }
+    public OAuthClientsAPI(Main main){
+        super(main, RECIPE_ID.OAUTH.toString());
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+
+        boolean containsAllRequired = containsAllRequiredFields(input);
+        boolean containsMoreThanAllowed = containsMoreThanAllowed(input);
+        if(!containsAllRequired || containsMoreThanAllowed){
+            throw new ServletException(new WebserverAPI.BadRequestException("Invalid Json Input"));
```

Comment:
need to give more details about whats invalid. From a dev's point of view, it will be a very bad experience to get such a vague error message

Reply:
Okay, I will change it to be more specific

Reply:
done. In the error message I list all the missing required fields. I decided that its not required to check for the `allowed` fields as the core just passes them through to the oauth provider, which ignores the fields that are not needed.